src/pubsub/ua_pubsub_networkmessage.c

```diff
@@ -768,6 +852,15 @@ UA_NetworkMessage_calcSizeBinary(UA_NetworkMessage *p, UA_NetworkMessageOffsetBu
         if(p->networkMessageType == UA_NETWORKMESSAGE_DATASET) {
             size += UA_Byte_calcSizeBinary(&p->payloadHeader.dataSetPayloadHeader.count);
             if(p->payloadHeader.dataSetPayloadHeader.dataSetWriterIds != NULL) {
+                if(offsetBuffer){
+                    if(offsetBuffer->RTsubscriberEnabled) {
+                        size_t pos = offsetBuffer->offsetsSize;
+                        if(!increaseOffsetArray(offsetBuffer))
+                            return 0;
```

Comment:
Return 0 is the same like UA_STATUSCODE_GOOD. This seems wrong here.

Reply:
This function returns the size of the total buffer so we should return 0 in case of failure.